SampleTest/App.xaml.cs

```diff
@@ -27,43 +31,65 @@ public App()
 
         // List of available authentication providers.
         private enum ProviderType
         {
             Mock,
-            Msal
+            Msal,
+            Windows
         }
 
         /// <summary>
         /// Initialize the global authentication provider.
         /// </summary>
-        private void InitializeGlobalProvider()
+        private async void InitializeGlobalProvider()
         {
             if (ProviderManager.Instance.GlobalProvider != null)
             {
                 return;
             }
 
-            // Provider config
-            string clientId = "YOUR_CLIENT_ID_HERE";
-            string[] scopes = { "User.Read", "User.ReadBasic.All", "People.Read", "Calendars.Read", "Mail.Read", "Group.Read.All", "ChannelMessage.Read.All" };
-
-            switch(_providerType)
+            await Window.Current.Dispatcher.TryRunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, async () =>
             {
-                // Mock provider
-                case ProviderType.Mock:
-                    ProviderManager.Instance.GlobalProvider = new MockProvider(signedIn: true);
-                    break;
-
-                //Msal provider
-                case ProviderType.Msal:
-                    ProviderManager.Instance.GlobalProvider = new MsalProvider(clientId, scopes);
-                    break;
-            }
+                // Provider config
+                string clientId = "YOUR_CLIENT_ID_HERE";
+                string[] scopes = { "User.Read", "User.ReadBasic.All", "People.Read", "Calendars.Read", "Mail.Read", "Group.Read.All", "ChannelMessage.Read.All" };
+
+                switch (_providerType)
+                {
+                    // Mock provider
+                    case ProviderType.Mock:
+                        ProviderManager.Instance.GlobalProvider = new MockProvider(signedIn: true);
+                        break;
+
+                    // Msal provider
+                    case ProviderType.Msal:
+                        var msalProvider = new MsalProvider(clientId, scopes);
+                        ProviderManager.Instance.GlobalProvider = msalProvider;
+                        await msalProvider.TrySilentSignInAsync();
+                        break;
+
+                    // Windows provider
+                    case ProviderType.Windows:
+                        var settingsCommandId = Guid.NewGuid();
+                        WindowsProvider windowsProvider = new WindowsProvider(clientId, scopes, new AccountsSettingsPaneConfig()
+                        {
+                            HeaderText = "Custom header text goes here.",
+                            Commands = new List<SettingsCommand>() { new SettingsCommand(settingsCommandId, "Click me!", OnSettingsCommandInvoked) }
+                        });
+                        ProviderManager.Instance.GlobalProvider = windowsProvider;
+                        await windowsProvider.TrySilentLoginAsync();
```

Comment:
Consider calling silent login automatically (for both all providers) when a provider is instantiated - mgt does the same and it removes one additional step from the developer. If a developer doesn't want to auto sign in, they can instantiate the provider when they want to attempt auto sign in.

If, however, this step is required, consider naming the method the same for the msal provider and the windows provider.

Reply:
Good idea, I can add that

Reply:
Thinking about this more, I like the idea of making it more simple but I think it should still be configurable. Folks have different paradigms and may want to be able to create the instance without logging in until they are ready. What do you think about adding another param to the constructor?

```
public WindowsProvider(...existingParams, autoLogin = true) { }
```

Also, I think the MsalProvider should be updated to use `TrySilentLoginAsync` instead of `TrySilentSignInAsync`. The other methods, `LoginAsync` and `LogoutAsync`, use `Log*` instead of `Sign*`. They should probably match.

Reply:
Hmm... the ProviderStates use `SignedIn`/`SignedOut`. And it seems MGT has the same inconsistency. Ok, I think they should match MGT after all.

Reply:
Up to you, whatever makes the most sense. FWIW, I don't think it's necessary to match how MGT does things if there is a better way to do it or if it makes more sense for Windows devs